src/plugins/workspace/public/components/workspace_detail/data_source_connection_table.tsx

```diff
@@ -27,26 +26,40 @@ import { i18n } from '@osd/i18n';
 import { DataSourceConnection, DataSourceConnectionType } from '../../../common/types';
 import { AssociationDataSourceModalMode } from '../../../common/constants';
 import { DirectQueryConnectionIcon } from '../workspace_form';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { CoreStart } from '../../../../../core/public';
 
 interface DataSourceConnectionTableProps {
   isDashboardAdmin: boolean;
   connectionType: string;
   dataSourceConnections: DataSourceConnection[];
-  handleUnassignDataSources: (dataSources: DataSourceConnection[]) => Promise<void>;
+  handleUnassignDataSources: (dataSources: DataSourceConnection[]) => void;
+  onSelectItems?: (dataSources: DataSourceConnection[]) => void;
+  inCreatePage?: boolean;
 }
 
 export const DataSourceConnectionTable = ({
   isDashboardAdmin,
   connectionType,
   dataSourceConnections,
   handleUnassignDataSources,
+  onSelectItems,
+  inCreatePage = false,
 }: DataSourceConnectionTableProps) => {
   const [selectedItems, setSelectedItems] = useState<DataSourceConnection[]>([]);
   const [modalVisible, setModalVisible] = useState(false);
   const [popoversState, setPopoversState] = useState<Record<string, boolean>>({});
   const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<
     Record<string, React.ReactNode>
   >({});
+  const {
+    services: { http },
+  } = useOpenSearchDashboards<CoreStart>();
+  useEffect(() => {
+    if (onSelectItems) {
+      onSelectItems(selectedItems);
+    }
+  }, [selectedItems, onSelectItems]);
```

Comment:
You can have this logic in `onSelectionChange` then you don't need this `useEffect`, then everywhere currently calls `setSelectedItems`, you can use `onSelectionChange`

Reply:
ok, thx!